.github/workflows/build-docker-image.yml

```
@@ -68,6 +68,7 @@ jobs:
       - name: Remove leading v from the version
         run: |
           echo "VERSION=$(echo ${{ github.ref_name }} | sed 's/^v//')" >> $GITHUB_ENV
+          echo "MAJOR_VERSION=$(echo ${{ github.ref_name }} | sed -E 's/^v([0-9]+)\.([0-9]+)\..+/\1.\2/')" >> $GITHUB_ENV
```

Useless use of `echo`. And actually, it is not a _major_ version you get, but major and minor version (or simply a minor version), OFC only if you follow semantic versioning.

```suggestion
          echo "MAJOR_VERSION=$(sed -E 's/^v([0-9]+\.[0-9]+).*/\1/')" <<< ${{github.ref_name}} >> $GITHUB_ENV
```

Or without `-E`:

```suggestion
          echo "MAJOR_VERSION=$(sed 's/^v\([0-9]\+\.[0-9]\+\).*/\1/')" <<< ${{github.ref_name}} >> $GITHUB_ENV
```

---

I could simplify. I'm not concerned about the major version name. Went with a few terms and then decided major and minor was too wordy.